### PR TITLE
Add functional methods for disks and Disk class

### PR DESCRIPTION
**Pull request by mziccard**
This PR also includes integration tests for disks images and snapshots.

### CODE REVIEW

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 1]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+    assertNull(compute.get(snapshotDiskId));
+    operation = snapshot.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.getSnapshot(snapshotName));
+  }
+
+  @Test
+  public void testListDisksAndSnapshots() throws InterruptedException {
```
Comment (mziccard) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Codacy Issue found: [The method testListDisksAndSnapshots() has an NPath complexity of 256](https://www.codacy.com/app/mziccard/gcloud-java/file/2372551756/issues/source?bid=3140968&fileBranchId=3223208#l1066)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 2]
```diff
@@ -918,6 +918,54 @@ public static ImageFilter notEquals(ImageField field, long value) {
...
+    }
+
+    /**
+     * Returns an equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static DiskFilter equals(DiskField field, String value) {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Codacy Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2372551672/issues/source?bid=3140968&fileBranchId=3223208#l938)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 3]
```diff
@@ -918,6 +918,54 @@ public static ImageFilter notEquals(ImageField field, long value) {
...
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static DiskFilter notEquals(DiskField field, String value) {
+      return new DiskFilter(checkNotNull(field), ComparisonOperator.NE, checkNotNull(value));
+    }
+
+    /**
+     * Returns an equals filter for the given field and long value.
+     */
+    public static DiskFilter equals(DiskField field, long value) {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Codacy Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2372551672/issues/source?bid=3140968&fileBranchId=3223208#l956)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 4]
```diff
@@ -1161,6 +1201,124 @@ public Operation deprecate(ImageId image,
...
+      }
+    };
+  }
+
+  @Override
+  public Page<Disk> listDisks(String zone, DiskListOption... options) {
+    return listDisks(zone, options(), optionMap(options));
+  }
+
+  private static Page<Disk> listDisks(final String zone, final ComputeOptions serviceOptions,
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'listDisks(String,ComputeOptions,Map)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2372551675/issues/source?bid=3140968&fileBranchId=3223208#l1254)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 5]
```diff
@@ -1161,6 +1201,124 @@ public Operation deprecate(ImageId image,
...
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Page<Disk> listDisks(DiskAggregatedListOption... options) {
+    return listDisks(options(), optionMap(options));
+  }
+
+  private static Page<Disk> listDisks(final ComputeOptions serviceOptions,
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'listDisks(ComputeOptions,Map)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2372551675/issues/source?bid=3140968&fileBranchId=3223208#l1282)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+  }
+
+  /**
+   * Creates a snapshot for this disk given the snapshot's name.
+   *
+   * @return a zone operation if the snapshot creation was successfully requested. {@code null} if
+   *     the disk was not found
+   * @throws ComputeException upon failure
+   */
+  public Operation createSnapshot(String snapshot, OperationOption... options) {
```
Comment (aozarov) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java]: I think I missed that in the compute.create(SnapshotInfo). Any reason to change the pattern and return `null` in create in this case?

**Comment by aozarov** [comment]
Looks good. I only have one question. @ajkannan do you mind to give it another pass and especially look at the tests/docs (which I completely ignored). If not, I will probably have time for a more thorough review tomorrow afternoon.

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java (hunk 6 above)
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java]: Also @ajkannan had noticed this before. The reason is that `createSnapshot` is an API of disk resources. Usually to create a resource we use the following endpoint (the example shows a disk):
```
POST /projects/<project>/zones/<zone>/disks/<disk>/insert
```
To create a snapshot we don't have an `/insert` endpoint, instead the endpoint is:
```
POST /projects/<project>/zones/<zone>/disks/<disk>/createSnapshot
```
Usually requests to `/insert` never return 404, `/createSnapshot` instead returns 404 if the disk does not exist. If you prefer for consistency we can avoid catching the 404 in this case.

Comment (aozarov) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java]: What would happen if zone not found in either of your examples?

**Comment by ajkannan** [comment]
Yes, I can give this a pass now.

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 7]
```diff
@@ -1586,6 +1634,110 @@ public static ImageListOption fields(ImageField... fields) {
...
+
+    private static final long serialVersionUID = -4354796876226661667L;
+
+    private DiskOption(ComputeRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the disk's fields to be returned by the RPC call. If this option
+     * is not provided, all disk's fields are returned. {@code DiskOption.fields} can be used to
```
Comment (ajkannan) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: nit: s/all disk's fields/all disk fields/

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 8]
```diff
@@ -1586,6 +1634,110 @@ public static ImageListOption fields(ImageField... fields) {
...
+     * specify only the fields of interest. {@link Disk#diskId()},
+     * {@link DiskConfiguration#diskType()} and {@link SnapshotDiskConfiguration#sourceSnapshot()}
```
Comment (ajkannan) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Nit: could add "either" after "and" to resolve a tiny bit of grammatical ambiguity.

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 9]
```diff
@@ -1586,6 +1634,110 @@ public static ImageListOption fields(ImageField... fields) {
...
+    /**
+     * Returns an option to specify the page token from which to start listing disks.
+     */
+    public static DiskListOption pageToken(String pageToken) {
+      return new DiskListOption(ComputeRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option to specify the disk's fields to be returned by the RPC call. If this option
+     * is not provided, all disk's fields are returned. {@code DiskListOption.fields} can be used to
```
Comment (ajkannan) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Could apply same nits as above (disk's --> disk, add "either" after "and" in second to last line)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java (hunk 7 above)
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: The saxon genitive is used for all `field` selector added so far... do you feel strongly about this?:)

Comment (ajkannan) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: It can stay :)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 10]
```diff
@@ -1161,6 +1201,124 @@ public Operation deprecate(ImageId image,
...
+            public com.google.api.services.compute.model.Operation call() {
+              return computeRpc.createDisk(disk.diskId().zone(), diskPb, optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER));
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  private static Function<com.google.api.services.compute.model.Disk, Disk> diskFromPb(
```
Comment (ajkannan) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Just curious, any reason to put this method here rather than having a package-scope static `Function` in the Disk class (like in Storage's Blob?).  If a user is using a lot of paging and list calls (perhaps an uncommon use case), could it be slightly more efficient to have the `Function` declared once statically in Disk rather than returning a new `Function` each time?  I don't feel strongly about changing it, I'm fine with keeping it as is if you prefer that.

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+   */
+  public static class Builder extends DiskInfo.Builder {
+
+    private final Compute compute;
+    private final DiskInfo.BuilderImpl infoBuilder;
+
+    Builder(Compute compute, DiskId diskId, DiskConfiguration diskConfiguration) {
+      this.compute = compute;
+      this.infoBuilder = new DiskInfo.BuilderImpl(diskId, diskConfiguration);
+      this.infoBuilder.diskId(diskId);
```
Comment (ajkannan) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java]: Is this line and the line below unnecessary because `diskId` and `diskConfiguration` are constructor arguments?

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+   */
+  public Operation createImage(String image, String description, OperationOption... options) {
+    ImageInfo imageInfo = ImageInfo.builder(ImageId.of(image), DiskImageConfiguration.of(diskId()))
+        .description(description)
+        .build();
+    return compute.create(imageInfo, options);
+  }
+
+  /**
+   * Returns the disk's {@code Compute} object used to issue requests.
```
Comment (ajkannan) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java]: nit: If you want, you could do a `@link` here

**Comment by ajkannan** [comment]
Question: is [disk resizing](https://cloud.google.com/compute/docs/reference/latest/disks/resize) something that we don't want to implement right now?

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/DiskTest.java [hunk 13]
```diff
@@ -0,0 +1,452 @@
...
+  private static final DiskTypeId TYPE = DiskTypeId.of("project", "zone", "disk");
+  private static final List<LicenseId> LICENSES = ImmutableList.of(
+      LicenseId.of("project", "license1"), LicenseId.of("project", "license2"));
+  private static final List<InstanceId> ATTACHED_INSTANCES = ImmutableList.of(
+      InstanceId.of("project", "zone", "instance1"),
+      InstanceId.of("project", "zone", "instance2"));
+  private static final SnapshotId SNAPSHOT = SnapshotId.of("project", "snapshot");
+  private static final ImageId IMAGE = ImageId.of("project", "image");
+  private static final String SNAPSHOT_ID = "snapshotId";
+  private static final String IMAGE_ID = "snapshotId";
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/DiskTest.java]: s/snapshotId/imageId/

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 14]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
   }
+
+  @Test
+  public void testCreateGetAndDeleteStandardDisk() throws InterruptedException {
+    String name = BASE_RESOURCE_NAME + "create-and-get-standard-disk";
+    DiskId diskId = DiskId.of(ZONE, name);
+    DiskInfo diskInfo =
+        DiskInfo.of(diskId, StandardDiskConfiguration.of(DiskTypeId.of(ZONE, "pd-ssd"), 100L));
+    Operation operation = compute.create(diskInfo);
+    while (!operation.isDone()) {
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: This might be a silly question, but does the Operation status "done" correspond to the Disk status "ready"?

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 15]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+    assertEquals(diskId.disk(), remoteDisk.diskId().disk());
+    assertEquals(DiskInfo.CreationStatus.READY, remoteDisk.creationStatus());
+    assertNotNull(remoteDisk.creationTimestamp());
+    assertNotNull(remoteDisk.id());
+    assertTrue(remoteDisk.configuration() instanceof ImageDiskConfiguration);
+    assertEquals(IMAGE_ID, remoteDisk.<ImageDiskConfiguration>configuration().sourceImage());
+    assertNotNull(remoteDisk.<ImageDiskConfiguration>configuration().sourceImageId());
+    assertEquals(DiskConfiguration.Type.IMAGE, remoteDisk.configuration().type());
+    assertNotNull(remoteDisk.configuration().sizeGb());
+    assertEquals("pd-standard", remoteDisk.configuration().diskType().diskType());
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: I think makes me wonder a little about diskType naming.  Could we make the string disk type inside DiskTypeId just "type"?

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 16]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+    }
+    // test get
+    Disk remoteDisk = compute.get(diskId);
+    assertNotNull(remoteDisk);
+    assertEquals(ZONE, remoteDisk.diskId().zone());
+    assertEquals(diskId.disk(), remoteDisk.diskId().disk());
+    assertEquals(DiskInfo.CreationStatus.READY, remoteDisk.creationStatus());
+    assertNotNull(remoteDisk.creationTimestamp());
+    assertNotNull(remoteDisk.id());
+    assertTrue(remoteDisk.configuration() instanceof ImageDiskConfiguration);
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Nit: if you want, you could save some configuration() calls and casting if you save remoteDisk.configuration() to a variable (here and for similar cases below).

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 17]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+    assertNotNull(snapshot.creationTimestamp());
+    assertNull(snapshot.description());
+    assertNull(snapshot.status());
+    assertNull(snapshot.diskSizeGb());
+    assertNull(snapshot.licenses());
+    assertNull(snapshot.sourceDisk());
+    assertNull(snapshot.sourceDiskId());
+    assertNull(snapshot.storageBytes());
+    assertNull(snapshot.storageBytesStatus());
+    SnapshotInfo.of(SnapshotId.of(snapshotName), snapshotDiskId);
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: What's this line's purpose?

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 18]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+      assertNull(image.deprecationStatus());
+    }
+  }
+
+  @Test
+  public void testListImagesWithFilter() {
+    Page<Image> imagePage = compute.listImages(IMAGE_PROJECT, Compute.ImageListOption.filter(
+        Compute.ImageFilter.equals(Compute.ImageField.ARCHIVE_SIZE_BYTES, 365056004L)));
+    Iterator<Image> imageIterator = imagePage.iterateAll();
+    while (imageIterator.hasNext()) {
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Do you think that we should assert that we got at least one result?  Suppose our filter construction was wrong, we could get no results, and this test would pass.

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 19]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+      assertNotNull(image.diskSizeGb());
+    }
+  }
+
+  @Test
+  public void testListImagesWithSelectedFields() {
+    Page<Image> imagePage =
+        compute.listImages(IMAGE_PROJECT, Compute.ImageListOption.fields(Compute.ImageField.ID));
+    Iterator<Image> imageIterator = imagePage.iterateAll();
+    while (imageIterator.hasNext()) {
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Do you think we should assert that this while loop is entered at least once?

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 20]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+      assertNull(remoteSnapshot.status());
+      assertNull(remoteSnapshot.diskSizeGb());
+      assertNull(remoteSnapshot.sourceDisk());
+      assertNull(remoteSnapshot.sourceDiskId());
+      assertNull(remoteSnapshot.storageBytes());
+      assertNull(remoteSnapshot.storageBytesStatus());
+      count++;
+    }
+    assertEquals(2, count);
+    compute.delete(firstDiskId);
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Here and in the method below you don't wait for the operation to complete before exiting the test, unlike in the methods above.  Any particular reason for that?

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 21]
```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+    ImageId imageId = ImageId.of(imageName);
+    DiskInfo diskInfo =
+        DiskInfo.of(diskId, StandardDiskConfiguration.of(DiskTypeId.of(ZONE, "pd-ssd"), 100L));
+    Operation operation = compute.create(diskInfo);
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    Disk remoteDisk = compute.get(diskId);
+    ImageInfo imageInfo = ImageInfo.of(imageId, DiskImageConfiguration.of(diskId));
+    operation = compute.create(imageInfo);
```
Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Did you also consider creating an Image using a StorageImageConfiguration?

**Comment by ajkannan** [comment]
Done with my pass.

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java (hunk 10 above)
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: The function in `Blob` takes a tuple and is only used in `BatchResponse submit(BatchRequest batchRequest)` to allow `transformBatchResult` to be more generic. The function takes a tuple, if we have a collection I think it will take more (both time and memory) to create a tuple `<Strorage, Disk>` for each item in the collection and then call the `Function` rather than having a factory method for the transformation function which we only need to call once for each collection transformation.

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java (hunk 12 above)
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java]: I didn't use `@link` because Compute should be already linked in the method's javadoc as the return type.

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java (hunk 14 above)
Comment (mziccard) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Yes the operation is done when the disk is READY and yes it is returned in CREATING status.

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java (hunk 15 above)
Comment (mziccard) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: I don't mind doing this change but I would leave it for a different PR.

Comment (ajkannan) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Sure, sounds good.  Maybe we can create an issue for it?

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java (hunk 20 above)
Comment (mziccard) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: I generally don't wait when the method is not testing `delete` and there are no constraint on the maximum number of that resources that can be created. This allows the tests to run faster.

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java (hunk 21 above)
Comment (mziccard) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Yes but it's not that easy. You need a raw image to be loaded in cloud storage. Images are quite big and we can not load them on the go in our ITs. Having the raw image loaded in one of our buckets makes the ITs non-repeatable by users. Given this I decided to skip that test.

**Comment by mziccard** [comment]
I fixed all comments. To add resize support I updated the compute dependency which also removed `MaintenanceWindow`'s list from `Zone`. Fixed that as well.

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java [hunk 22]
```diff
@@ -855,4 +874,486 @@ public void testListGlobalAddresses() throws InterruptedException {
...
+    assertNull(compute.get(snapshotDiskId));
+    operation = snapshot.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.getSnapshot(snapshotName));
+  }
+
+  @Test
+  public void testListDisksAndSnapshots() throws InterruptedException {
```
Comment (mziccard) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Codacy Issue found: [The method testListDisksAndSnapshots() has an NPath complexity of 256](https://www.codacy.com/app/mziccard/gcloud-java/file/2383968105/issues/source?bid=3140968&fileBranchId=3223208#l1073)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java (hunk 2 above)
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Codacy Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2383968021/issues/source?bid=3140968&fileBranchId=3223208#l938)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java (hunk 3 above)
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Codacy Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2383968021/issues/source?bid=3140968&fileBranchId=3223208#l956)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 23]
```diff
@@ -1161,6 +1201,141 @@ public Operation deprecate(ImageId image,
...
+      }
+    };
+  }
+
+  @Override
+  public Page<Disk> listDisks(String zone, DiskListOption... options) {
+    return listDisks(zone, options(), optionMap(options));
+  }
+
+  private static Page<Disk> listDisks(final String zone, final ComputeOptions serviceOptions,
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'listDisks(String,ComputeOptions,Map)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2383968024/issues/source?bid=3140968&fileBranchId=3223208#l1254)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 24]
```diff
@@ -1161,6 +1201,141 @@ public Operation deprecate(ImageId image,
...
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Page<Disk> listDisks(DiskAggregatedListOption... options) {
+    return listDisks(options(), optionMap(options));
+  }
+
+  private static Page<Disk> listDisks(final ComputeOptions serviceOptions,
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'listDisks(ComputeOptions,Map)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2383968024/issues/source?bid=3140968&fileBranchId=3223208#l1282)

File: gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java (hunk 15 above)
Comment (mziccard) [inline gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java]: Opened #887

**Comment by ajkannan** [comment]
`MAINTENANCE_WINDOW` still exists in the `ZoneField` enum (in `Compute.java`) -- is that intentional or should it be removed?

**Comment by mziccard** [comment]
> MAINTENANCE_WINDOW still exists in the ZoneField enum (in Compute.java) -- is that intentional or should it be removed?

I forgot, it should be removed.

**Comment by ajkannan** [comment]
LGTM, feel free to merge after removing the field from the ZoneFields.
